pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -1629,14 +1629,21 @@ void Object_beginTypeContext(JSOBJ _obj, JSONTypeContext *tc) {
     if (PyLong_Check(obj)) {
         PRINTMARK();
         tc->type = JT_LONG;
-        GET_TC(tc)->longValue = PyLong_AsLongLong(obj);
+        int overflow = 0;
+        GET_TC(tc)->longValue = PyLong_AsLongLongAndOverflow(obj, &overflow);
+        int err;
+        err = (GET_TC(tc)->longValue == -1) && PyErr_Occurred();
 
-        exc = PyErr_Occurred();
+        if (overflow){
+            PRINTMARK();
```

Comment:
I think we do need to add a `PyErr_Clear();` within this branch to prevent this from incorrectly propogating an OverflowError (sorry think I mistakenly asked you to remove before)

Reply:
I think that - if I understand the docstring correctly - `PyLong_AsLongLongAndOverflow` doesn't raise an OverflowError (but will raise an error if there's a problem other than Overflow)

One thing I know is this runs ok (including tests) without the `PyErr_Clear()` though I guess it doesn't hurt to add that in?

Reply:
Yea I think you are right. Let’s keep this as is